core/lib/src/request/from_request.rs

```diff
@@ -169,6 +169,12 @@ impl<S, E> IntoOutcome<S, (Status, E), ()> for Result<S, E> {
 ///     via [`Request::client_ip()`]. If the client's IP address is not known,
 ///     the request is forwarded.
 ///
+///   * **Protocol**
+///
+///     Extracts the protocol of the incoming request as a [`Protocol`] via
+///     [`Request::proxy_proto()`] (HTTP or HTTPS). If the used protocol is not
+///     known, the request is forwarded.
+///
```

Comment:
This seems problematic. I would expect such a guard to succeed 99.9% of the time, given the name of the type. But this will only work when the header is set. Is there a benefit to exposing this? If we really want to do this, we should change the name to something less general, like `ForwardedProtocol`, or something.

Reply:
While working on this I thought I would need this, but I realised that in the end we didn't but it might be useful to someone else. As it's trivial to implement, I will just remove it.

Reply:
Decided to keep it, as the tests I added are using it, but I renamed it to `ForwardedProtocol`.

Reply:
This discussion still stands, unless keeping it is now deemed acceptable. Also see https://github.com/rwf2/Rocket/pull/2519#discussion_r1428635476